lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1601,20 +1621,34 @@ bool ConstExprEvaluator::decodeAllocUninitializedArray(
           return true;
 
         index = ili->getValue().getLimitedValue();
-        user = iai->getSingleUserOfType<StoreInst>();
+        use = iai->getSingleUse();
+        user = use ? use->getUser() : nullptr;
       }
 
-      // Check to see if we have a store to a valid index that hasn't been
-      // stored to yet.
-      auto *store = dyn_cast_or_null<StoreInst>(user);
-      if (!store || index >= elements.size() || elements[index] != SILValue())
+      // We handle the cases that the element is either set by a store or
+      // filled via an apply_inst.
+      if (auto *store = dyn_cast_or_null<StoreInst>(user)) {
+        if (store->getDest() != use->get())
+          return true;
+      } else if (auto *applyInst = dyn_cast_or_null<ApplyInst>(user)) {
+        // Check to see if this is an out-parameter (i.e., like a store).
```

Comment:
can we add a SIL snippet here as a code comment to show what cases we are dealing with here?

Reply:
Added the `SignedInteger<>.init<A>(_:)` case